### PR TITLE
Added PR template, Security Policy and CODEOWNERS

### PR DESCRIPTION
**Pull request by mdawsonuk**
This PR adds the following features:
- A Pull Request template which requires users to certify that they have completed a set number of tasks
- A Security Policy should any security vulnerabilities be discovered, as well as a scope of vulnerabilities
- A Codeowners file - this currently defaults to all files being owned by Troy

## Checklist:

- [ ] I confirm that my submission adheres to the [Code of Conduct](https://github.com/HaveIBeenPwned/PwnedPasswordsAzureFunction/blob/main/CODE_OF_CONDUCT.md).
- [ ] I have performed a self-review of all of my code
- [ ] I have ensured all of my code has `<summary>` tags for all methods and public variables, and has comments to understand program execution flow where it is hard to understand, as well as updating any `<summary>` tags in functions where functionality has changed.
- [ ] I have tested that the code works with sample data running the code either locally or in my own deployed Azure Function.
- [ ] I have verified that any tests that I have supplied with this Pull Request work and I have fixed any code which caused pre-existing tests to fail.
- [ ] I have updated the [README](https://github.com/HaveIBeenPwned/PwnedPasswordsAzureFunction/blob/main/README.md) with any changes in dependencies/configuration values.
- [ ] I have linted my code to ensure that it is formatted correctly.

---

## Description

This adds a new Pull Request Template like this one which users fill out to a description of the changes as well as ensuring that a few basic tasks have been filled - such as ensuring their code doesn't break any tests, their code is properly commented etc. This also adds a Security Policy which gives users a point of contact, a policy to follow until the vulnerability is fixed, as well as a scope. Finally, a `CODEOWNERS` file has been added, with all files currently owned by Troy.

## Checklist:

- [x] I confirm that my submission adheres to the [Code of Conduct](https://github.com/HaveIBeenPwned/PwnedPasswordsAzureFunction/blob/main/CODE_OF_CONDUCT.md).
- [x] I have performed a self-review of all of my code
- [x] I have ensured all of my code has `<summary>` tags for all methods and public variables, and has comments to understand program execution flow where it is hard to understand, as well as updating any `<summary>` tags in functions where functionality has changed.
- [x] I have tested that the code works with sample data running the code either locally or in my own deployed Azure Function.
- [x] I have verified that any tests that I have supplied with this Pull Request work and I have fixed any code which caused pre-existing tests to fail.
- [x] I have updated the [README](https://github.com/HaveIBeenPwned/PwnedPasswordsAzureFunction/blob/main/README.md) with any changes in dependencies/configuration values.
- [x] I have linted my code to ensure that it is formatted correctly.

### CODE REVIEW

File: .github/SECURITY.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+- (Optional) Code, screenshots, or videos of the vulnerability (but no executable binaries)
+
+For sensitive communications, you can use [Keybase](https://keybase.io/troyhunt).
+
+## Scope
+
+The following items are **not** in scope:
+- High volume vulnerabilities, such as overwhelming the service with requests, Dos, brute force attacks, etc.
+- Rate limitations or service limitations, unless it allows bypassing of rate-limits, API keys, or other security measures
+- Vulnerabilities from old versions of the project
```
Comment (tomkerkhove) [inline .github/SECURITY.md:24]: Depending on what @troyhunt is planning, but I'd add this:
```suggestion
- Vulnerabilities from old versions of the project

## Compensation
We do not provide compensations for reporting vulnerabilities except for eternal gratitude.
```

Comment (tomkerkhove) [inline .github/SECURITY.md:24]: It should cover how communication will happen:
```suggestion
- Vulnerabilities from old versions of the project

## Communication
GitHub Security Advisory will be used to communicate during the process of identifying, fixing & shipping the mitigation of the vulnerability.

The advisory will only be made public when the patched version is released to inform the community of the breach and its potential security impact.
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+## Checklist:
+<!-- Please replace every instance of `[ ]` with `[X]` to indicate you have completed the criteria -->
+
+- [ ] I confirm that my submission adheres to the [Code of Conduct](https://github.com/HaveIBeenPwned/PwnedPasswordsAzureFunction/blob/main/CODE_OF_CONDUCT.md).
+- [ ] I have performed a self-review of all of my code
+- [ ] I have ensured all of my code has `<summary>` tags for all methods and public variables, and has comments to understand program execution flow where it is hard to understand, as well as updating any `<summary>` tags in functions where functionality has changed.
+- [ ] I have tested that the code works with sample data running the code either locally or in my own deployed Azure Function.
+- [ ] I have verified that any tests that I have supplied with this Pull Request work and I have fixed any code which caused pre-existing tests to fail.
+- [ ] I have updated the [README](https://github.com/HaveIBeenPwned/PwnedPasswordsAzureFunction/blob/main/README.md) with any changes in dependencies/configuration values.
+- [ ] I have linted my code to ensure that it is formatted correctly.
```
Comment (tomkerkhove) [inline .github/PULL_REQUEST_TEMPLATE.md:18]: Experience has learned me that your PR templates should be short or nobody will actually bother, so I'd propose the following:

```suggestion
- [ ] I confirm that my submission adheres to the [Code of Conduct](https://github.com/HaveIBeenPwned/PwnedPasswordsAzureFunction/blob/main/CODE_OF_CONDUCT.md).
- [ ] I have performed a self-review of all of my code
- [ ] I have tested that the code works with sample data and verified all tests are passing
- [ ] I have updated the [README](https://github.com/HaveIBeenPwned/PwnedPasswordsAzureFunction/blob/main/README.md) with any changes in dependencies/configuration values.
- [ ] I have linted my code to ensure that it is formatted correctly.
```

**Review by tomkerkhove — APPROVED** [review]

**Comment by stebet** [comment]
I'd suggest making the code documentation optional, especially for private/internal methods. In my experience the code docs really only make sense for a public API or library referenced in other projects and often just convolute the code files with boilerplate text when it is perfectly understandable without them.

**Comment by mdawsonuk** [comment]
> I'd suggest making the code documentation optional, especially for private/internal methods. In my experience the code docs really only make sense for a public API or library referenced in other projects and often just convolute the code files with boilerplate text when it is perfectly understandable without them.

I've removed the item entirely from the checklist

**Comment by mdawsonuk** [comment]
Any updates or changes for this PR?

**Comment by stebet** [comment]
> Any updates or changes for this PR?

I'll delegate the acceptance of this PR to @troyhunt since he's better suited to evaluate the contents of the security policy and the code ownership.

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -0,0 +1 @@
+*       @troyhunt
```
Comment (stebet) [inline .github/CODEOWNERS:1]:
```suggestion
*       @troyhunt

```

Comment (stebet) [reply .github/CODEOWNERS:1]: Adding a newline at the end of the file

**Review by stebet — CHANGES_REQUESTED** [review]

**Pull request merged**